core/consensus/msg.go

```diff
@@ -112,10 +119,6 @@ func (m msg) ToConsensusMsg() *pbv1.ConsensusMsg {
 
 // hashProto returns a deterministic ssz hash root of the proto message.
 func hashProto(msg proto.Message) ([32]byte, error) {
-	if msg == nil {
-		return [32]byte{}, nil
-	}
```

Comment:
Checking if an interface is nil is bad, since nil pointer implementations of the interface isn't nil...